### ISSUE TITLE
a strange warnning when running cf

### ISSUE THREAD

**Issue by cypro666** [mlpack/mlpack#598]
./mlpack_cf -t train.csv --output_file model.csv
[WARN ] Number of random columns (columnsToAverage) is more than the number of columns available in the V matrix; weird results may ensue!

the train.csv is:
0,1,4
1,0,5
1,1,1
2,0,2

**Comment by rcurtin** [comment]
This warning will be issued whenever the data matrix has fewer than five points, because the matrix decomposition is (by default) initialized using 5 random columns of the training data.  With any larger data matrix this will not be a problem.  In your case I am guessing that you are just testing the program, so I would not worry about the warning.  If you are getting weird results like the warning suggests, maybe adding a couple more points to your dataset would be useful.

**Comment by cypro666** [comment]
thx， I‘ve found the reason made this warn, and another little stuff:
./mlpack_det
terminate called after throwing an instance of 'std::runtime_error'
  what():  fatal error; see Log::Fatal output
Aborted (core dumped)

If No arguments applied, I think help info should be show, boost::program_options support this util
, cause `Aborted ` if not very friendly :)

**Comment by rcurtin** [comment]
Are you sure it's displaying nothing at all?  It should be displaying an error too, not just the exception.  Here is what I get on my system (compiled without debugging symbols):

```
$ bin/mlpack_det
[FATAL] Neither --training_file (-t) nor --input_model_file (-m) are specified!

terminate called after throwing an instance of 'std::runtime_error'
  what():  fatal error; see Log::Fatal output
Aborted
```

There was an issue with this output when compiled with debugging symbols, which I fixed in 78f53da.  Do you want to update to the latest git master revision and see what you get as output?  Thanks!

**Comment by cypro666** [comment]
thanks, fixed in 78f53da
